dll/cpl/sysdm/environment.c

```diff
@@ -353,6 +384,10 @@ EditVariableDlgProc(HWND hwndDlg,
             /* Either get the values from list box or from edit box */
             if (DlgData->bIsFriendlyUI)
             {
+                /* Subclass the list box control first */
+                DlgData->OldListBoxProc = (WNDPROC)GetWindowLongPtr(GetDlgItem(hwndDlg, IDC_LIST_VARIABLE_VALUE), GWLP_WNDPROC);
```

Comment:
`SetWindowSubclass` is safer, and handles complex cases better (for example where multiple subclasses are being set on the same window.)